src/util.ts

```diff
@@ -2,7 +2,13 @@ export type DispatchOptions = { target: EventTarget, cancelable: boolean, detail
 
 export function dispatch(eventName: string, { target, cancelable, detail }: Partial<DispatchOptions> = {}) {
   const event = new CustomEvent(eventName, { cancelable, bubbles: true, detail })
-  void (target || document.documentElement).dispatchEvent(event)
+
+  if (target && document.body.contains(target as Element)) {
```

Comment:
Would [Node.isConnected](https://developer.mozilla.org/en-US/docs/Web/API/Node/isConnected) be an interchangeable check here?

Reply:
Good call. Works on shadow doms as well.